LoadGen: API and source documentation

briandersn opened the pull request.

briandersn commented:
Hi @christ1ne ! Ptal.

Btw, is there somewhere we can host the generated html documentation for easy viewing without having to check out the the repository?

christ1ne commented:
not sure.. maybe we can ask at the next inference group about where to host the doxygen documentation.

briandersn commented:
Ok. I removed the generated html documentation for now. I've also rebased on top of Anton's recent clang warnings fix, which was conflicting.

This is a pure documentation change, so I'm hoping to land it soon for the benefit of all the users and contributors.

Thanks!
